addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -283,6 +284,42 @@ def test_on_next_order_reward_promotion_program_with_requirements(self):
         order_bis._update_programs_and_rewards()
         self.assertEqual(len(order_bis.order_line), 2, "Free product from a coupon generated from a promotion program on next order should not dissapear")
 
+    def test_partner_assigned_to_next_order_coupon(self):
+        """ Test the assignment of a partner on coupons with program type `next_order_coupons`.
+
+        1. Create a loyalty program of type `next_order_coupons`.
+        2. Create a sale order and add a product to it.
+        3. Apply the loyalty program to the sale order.
+        4. Verify that the generated coupon is assigned to the order's partner.
+        """
+        loyalty_program = self.env['loyalty.program'].create({
```

Comment:
If we force a partner on all the orders containing a named program, this test should be in the frontend.

Reply:
This test case is particularly for `next_order_coupons` only while the remaining named programs like `loyalty cards` and `e-wallet` where the partner is needed, are not covered by this test case.